packages/node_modules/@ciscospark/spark-core/src/lib/spark-core-plugin-mixin.js

```diff
@@ -58,15 +58,17 @@ export default function mixinSparkCorePlugins(State, config, interceptors) {
       }));
     }
 
-
-    if (constructor.prototype._definition.ready) {
-      const {fn, depList} = State.prototype._derived.ready;
-      const def = {
-        deps: depList.concat(`${name}.ready`),
-        fn
-      };
-
-      createDerivedProperty(State.prototype, 'ready', def);
+    // Only mess with the plugin's derived properties if it's an amp-state plugin
+    if (constructor.prototype._definition) {
+      if (constructor.prototype._definition.ready) {
```

Comment:
Could this be combined into:
``` js
if (constructor.prototype._definition &&
  constructor.prototype._definition.ready) {
```

Reply:
yea, it could be. I liked the semantic difference of "check if this is a SparkPlugin vs StatelessSparkPlugin", then check if it has a ready property. I can certainly change it.